core-web/libs/sdk/experiments/src/lib/dot-experiments.ts

```diff
@@ -197,20 +258,60 @@ export class DotExperiments {
      *
      * @private
      */
-    private async initAnalyticsClient() {
+    private initAnalyticsClient() {
         try {
             this.analytics = jitsuClient({
                 key: this.config['api-key'],
                 tracking_host: this.config['server'],
                 log_level: this.config['debug'] ? DEBUG_LEVELS.DEBUG : DEBUG_LEVELS.WARN
             });
 
-            const { experiments } = await this.getDataForAnalytics();
+            const { experiments } = parseDataForAnalytics(this.indexDBData.experiments, location);
             this.analytics.set({ experiments });
 
             dotLogger(`Analytics client created successfully.`, this.getIsDebugActive());
         } catch (error) {
             throw Error(`Error creating analytics client, ${error}`);
         }
     }
+
+    /**
+     * Determines whether analytics should be checked.
+     *
+     * @private
+     * @returns {Promise<boolean>} A boolean value indicating whether analytics should be checked.
+     */
+    private shouldCheckAnalytics(): boolean {
+        // If the user close the tab, reload the data
+        if (!checkFlagExperimentAlreadyChecked()) {
+            this.persistenceHandler.clearData();
+
+            return true;
+        }
+
+        if (!this.indexDBData) {
+            return true;
+        }
+
+        if (this.indexDBData && checkInvalidateDataChecked(this.indexDBData.created)) {
```

Comment:
checking for `this.indexDBData` is redundant

Reply:
moved the verification to the util.